apps/etterlatte-saksbehandling-ui/client/src/components/behandling/trygdetid/avtaler/TrygdeAvtale.tsx

```diff
@@ -0,0 +1,275 @@
+  TrygdetidAvtaleKriteria,
+  hentAlleTrygdetidAvtaler,
+  hentTrygdeavtaleForBehandling,
+  Trygdeavtale,
+  TrygdetidAvtaleOptions,
+  lagreTrygdeavtaleForBehandling,
+  TrygdeavtaleRequest,
+} from '~shared/api/trygdetid'
+import { isFailure, isPending, useApiCall } from '~shared/hooks/useApiCall'
+import { IconSize } from '~shared/types/Icon'
+import { Innhold, FormWrapper, FormKnapper } from '../styled'
+
+interface TrygdeavtaleVisningProps {
+  trygdeavtale: Trygdeavtale
+  avtaler: TrygdetidAvtale[]
+  kriterier: TrygdetidAvtaleKriteria[]
+}
+
+const TrygdeavtaleVisning = ({ trygdeavtale, avtaler, kriterier }: TrygdeavtaleVisningProps) => {
+  const avtale = avtaler.find((avtale) => trygdeavtale.avtaleKode === avtale.kode)
+  const avtaleDato = avtale?.datoer.find((dato) => trygdeavtale.avtaleDatoKode === dato.kode)
+  const avtaleKriteria = kriterier.find((kriteria) => trygdeavtale.avtaleKriteriaKode === kriteria.kode)
+
+  return (
+    <TrygdeavtaleBody>
+      {avtale && avtale.beskrivelse}
+      {avtaleDato && <> - {avtaleDato.beskrivelse}</>}
+      {avtaleKriteria && (
+        <>
+          <br />
+          {avtaleKriteria.beskrivelse}
+        </>
+      )}
+    </TrygdeavtaleBody>
+  )
+}
+
+const TrygdetidAvtaleOption: React.FC<{ defaultBeskrivelse: string; options: TrygdetidAvtaleOptions[] }> = ({
+  defaultBeskrivelse,
+  options,
+}) => {
+  return (
+    <>
+      <option value="">{defaultBeskrivelse}</option>
+      {options.map((opt) => (
+        <option key={`${opt.kode}`} value={opt.kode}>
+          {opt.beskrivelse}
+        </option>
+      ))}
+    </>
+  )
+}
+
+interface Props {
+  redigerbar: boolean
+}
+
+export const TrygdeAvtale = ({ redigerbar }: Props) => {
+  const { behandlingId } = useParams()
+  const [hentAlleTrygdetidAvtalerRequest, fetchTrygdetidAvtaler] = useApiCall(hentAlleTrygdetidAvtaler)
+  const [hentAlleTrygdetidAvtalerKriterierRequest, fetchTrygdetidAvtaleKriterier] = useApiCall(
+    hentAlleTrygdetidAvtaleKriterier
+  )
+  const [lagreTrygdeavtaleRequest, lagreTrygdeavtale] = useApiCall(lagreTrygdeavtaleForBehandling)
+  const [hentTrygdeavtaleRequest, fetchTrygdeavtale] = useApiCall(hentTrygdeavtaleForBehandling)
+  const [avtalerListe, setAvtalerListe] = useState<TrygdetidAvtale[]>()
+  const [avtaleKriterierListe, setAvtaleKriterierListe] = useState<TrygdetidAvtaleKriteria[]>()
+  const [trygdeavtale, setTrygdeavtale] = useState<Trygdeavtale>({} as Trygdeavtale)
+  const [valgtAvtale, setValgtAvtale] = useState<TrygdetidAvtale>()
+  const [redigering, setRedigering] = useState<Boolean>(true)
+  const [harLagretVerdi, setHarLagretVerdi] = useState<Boolean>(false)
+
+  useEffect(() => {
+    fetchTrygdetidAvtaler(null, (avtaler: TrygdetidAvtale[]) => {
+      setAvtalerListe(avtaler.sort((a: TrygdetidAvtale, b: TrygdetidAvtale) => trygdeavtaleOptionSort(a, b)))
+    })
+  }, [])
+
+  useEffect(() => {
+    fetchTrygdetidAvtaleKriterier(null, (avtaler: TrygdetidAvtaleKriteria[]) => {
```

Comment:
Dette kan du gjøre i den første useEffect da de skal trigges på det samme, unødvendig med to.

Reply:
Gjort. Var usikker på sep of concerns (en eller flere)